docs/_packages/web_api.md

```diff
@@ -466,68 +466,50 @@ retrying the API call. If you'd like to opt out of that behavior, set the `rejec
+Blobs of binary data, sometimes structured, sometimes called a "file", can be useful to upload and share to a channel.
+Such files can be uploaded and shared to a channel with the nifty `filesUploadV2` API:
 
-* Call WebClient#files.getUploadURLExternal() method to receive a URL to use for each file
-* Perform an HTTP POST request to the URL you received in step 1 for each file
-* Call WebClient#files.completeUploadExternal() method with the pairs of file ID and title to complete the whole process, plus share the files in a channel
-* If you would like the full metadata of the files, call WebClient#files.info() method for each file
-
-We understand that writing the above code requires many lines of code. Also, existing WebClient#files.upload() users have to take a certain amount of time for migration. To mitigate the pain, we've added a wrapper method named WebClient#files.uploadV2().
-
-Also, in addition to the performance improvements, another good news is that 3rd party apps can now upload multiple files at a time!
-
-See the following code examples demonstrating how the wrapper method works:
-
-##### Legacy way
 ```javascript
 const { WebClient } = require('@slack/web-api');
 
 const token = process.env.SLACK_TOKEN;
 const web = new WebClient(token);
 
-const result = await web.files.upload({
-  file: './path/to/logo.png',  // also accepts Buffer or ReadStream
+const result = await web.filesUploadV2({
+  channel_id: 'C0123456789',
+  initial_comment: 'Here is the new company logo!',
+  file: './path/to/logo.png',
   filename: 'logo.png',
-  channels: 'C12345',
-  initial_comment: 'Here is the new company logo',
 });
 
-// `result` contains information about the uploaded file
-console.log('File uploaded: ', result.file.id);
+console.log('File uploaded:', result.files);
 ```
 
-#### New way
-```javascript
-const { WebClient } = require('@slack/web-api');
+The `file` parameter accepts either the path to a file, a Buffer, or a ReadStream. Details on handling different forms
+of binary data follow soon!
 
-const token = process.env.SLACK_TOKEN;
-const web = new WebClient(token);
-const result = await web.files.uploadV2({
-  file: './path/to/logo.png',  // also accepts Buffer or ReadStream
-  filename: 'logo.png',
-  // Note that channels still works but going with channel_id="C12345" is recommended.
-  // channels="C111,C222" is no longer supported. In this case, an exception will be thrown
-  channels: 'C12345',
-  initial_comment: 'Here is the new company logo',
-});
+The `channel_id` and `initial_comment` aren't required, but the file either won't be shared to a channel or it won't be
+posted with a message if these aren't included.
 
-// `result may contain multiple files uploaded
-console.log('File(s) uploaded: ', result.files);
-```
+In a successful response, the `result.files` contains an array of [shared files](https://api.slack.com/types/file).
```

Comment:
The sharing with these APIs is tricky. If the user explicitly doesn't provide a `channel_id`, then the uploaded files will not be shared by default and will be 'private' - that is, only the owner of the token used to upload the file(s) will have access to the file(s), programmatically or in the client or otherwise.